### PR TITLE
style: updated global header to design specs

### PR DESCRIPTION
**Pull request by josephgregoryii**
Styling changes for updated global header designs.

### Desktop
Lightmode

Darkmode

### Mobile

Lightmode

Darkmode

resolves: #510

### CODE REVIEW

File: packages/gatsby-theme-newrelic/src/components/Button.js
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/Button.js:52]: I'm curious, what is a "locale" button? Is this only for the global header? If so, its worth considering making this a one-off there instead of a global type. Then again I have no context for these changes, so feel free to ignore me 😄

File: packages/gatsby-theme-newrelic/src/components/GlobalHeader.js
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/GlobalHeader.js:308]: Since this is now non-zero, this should have a unit associated with it.

File: packages/gatsby-theme-newrelic/src/components/GlobalHeader.js
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/GlobalHeader.js:348]:
```suggestion
                padding: 0 1rem;
```

We like to use rems for text. Let's update this to use rems as well (`1rem == 16px`)

File: packages/gatsby-theme-newrelic/src/components/GlobalHeader.js
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/GlobalHeader.js:433]:
```suggestion
                  margin: 0 0.625rem;
```

We like to use `rem` for margins. Let's update to use `rem` here.

File: packages/gatsby-theme-newrelic/src/components/GlobalNavLink.js
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/GlobalNavLink.js:38]:
```suggestion
        padding: 0.25rem 0.625rem;
```

Huh, not sure why we mixed units here. Could we use rem for padding instead?

File: packages/gatsby-theme-newrelic/src/components/GlobalNavLink.js
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/GlobalNavLink.js:52]:
```suggestion
        &:hover {
```

This looks to be missing the `&`

File: packages/gatsby-theme-newrelic/src/components/NewRelicLogo.js
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/NewRelicLogo.js:7]: Does this color work for both light and dark mode?

File: packages/gatsby-theme-newrelic/src/components/SearchInput.js
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/SearchInput.js:101]:
```suggestion
            border: 1px solid transparent;
```

I see that a border is added on hover. By reducing this to `0`, I fear this will make the surrounding UI jump since it goes from no border to a 1px border around it. Rather than removing the border in the default state, instead let's change the border color to transparent so that it takes up the same amount of room and avoids jumping around.

File: packages/gatsby-theme-newrelic/src/components/SearchInput.js
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/SearchInput.js:112]: These changes will affect all search inputs. Just to be sure, are you meaning to update this for all search inputs or just the search input in the global header?

File: packages/gatsby-theme-newrelic/src/components/SearchInput.js
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/SearchInput.js:103]: Does this work for both light and dark modes? This `SearchInput` is used globally, not just the header.

File: packages/gatsby-theme-newrelic/src/components/SearchInput.js
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/SearchInput.js:124]:
```suggestion
              border-color: var(--border-color);
```

Assuming the change is made above to use a transparent border color, you can instead just update the border color here.

File: packages/gatsby-theme-newrelic/src/components/Button.js
Comment (josephgregoryii) [reply packages/gatsby-theme-newrelic/src/components/Button.js:52]: The "locale" button is intended for the dropdown section in the header for language preference. Assuming that we are going to be adding more languages in the future, there may be multiple options where adding a  "locale" button variant may be beneficial. 

I'm not particularly attached to this idea, so I'm open to suggestions and happy to change this.

File: packages/gatsby-theme-newrelic/src/components/SearchInput.js
Comment (josephgregoryii) [reply packages/gatsby-theme-newrelic/src/components/SearchInput.js:101]: @zstix and I were trying to figure out where this was occurring. Thanks for that!

**Review by zstix — CHANGES_REQUESTED** [review]
This looks great! We had a few minor comments.

One other thing, which we spoke out on the call is that the header can get into this weird state at some odd breakpoints:

Screen Shot 2021-11-22 at 11 01 26

File: packages/gatsby-theme-newrelic/src/components/GlobalHeader.js
Comment (zstix) [inline packages/gatsby-theme-newrelic/src/components/GlobalHeader.js:397]: I noticed that the placeholder text "wiggles" a little bit when hovering over the input. This is (probably) because the border is pushing text over. If we want to avoid this, we could use an [inset `box-shadow`](https://developer.mozilla.org/en-US/docs/Web/CSS/box-shadow) to achieve the same effect:
```suggestion
                      box-shadow: inset 0 0 0 2px var(--hover-border-color);
}
```

**Review by zstix — DISMISSED** [review]

**Comment by josephgregoryii** [comment]
Sent a link for Àngel's approval. Once reviewed i'll merge once we get the green light.

File: packages/gatsby-theme-newrelic/src/components/GlobalHeader.js
Comment (aswanson-nr) [inline packages/gatsby-theme-newrelic/src/components/GlobalHeader.js:315]:
```suggestion
                @media screen and (max-width: ${MOBILE_BREAKPOINT}) {
```

File: packages/gatsby-theme-newrelic/src/components/GlobalHeader.js
Comment (aswanson-nr) [inline packages/gatsby-theme-newrelic/src/components/GlobalHeader.js:427]: I noticed this `545px` value is used in a few places. It would be nice to pop this out to a variable which would make it easier to change later and remove the "magic number"-ness by giving it a name.

File: packages/gatsby-theme-newrelic/src/components/GlobalHeader.js
Comment (aswanson-nr) [inline packages/gatsby-theme-newrelic/src/components/GlobalHeader.js:506]: Here is another place using the `545px` value

File: packages/gatsby-theme-newrelic/src/components/GlobalHeader.js
Comment (aswanson-nr) [inline packages/gatsby-theme-newrelic/src/components/GlobalHeader.js:539]: If possible, we could move this over to the `css` prop provided by emotion. Similar to how we moved the `SearchInput` over.

File: packages/gatsby-theme-newrelic/src/components/NewRelicLogo.js
Comment (aswanson-nr) [inline packages/gatsby-theme-newrelic/src/components/NewRelicLogo.js:14]: I think this is a similar situation where we could make use of the `css` property provided by emotion to do this in a "css"-y way.

**Review by aswanson-nr — CHANGES_REQUESTED** [review]
Looking good! Found a few more spots where the CSS can be pulled into the `css` prop provided by Emotion.

**Review by aswanson-nr — APPROVED** [review]
:shipit:

**Pull request merged**
